### ISSUE TITLE
Testing the WebAssembly components and service

### ISSUE THREAD

**Issue by pmarangoni** [markjprice/cs12dotnet8#60]
1. I run both **Northwind.WebApi** and **Northwind.Blazor** and open the browser's developer tools.
2. I navigate to https://localhost:5161/ and view the Console tab, and it is blank, unlike your description in step 5 which states: _"note that Blazor WebAssembly has loaded .NET assemblies into the browser cache and that they take about 10.65 MB of space, as shown in Figure 16.7"_
3. Image 16.7 also shows a reference to **blazor.webassembly.js**, but my browser's sources tab doesn't show that file anywhere. Something is clearly wrong here.
4. I can load customers when viewing **Customers Worldwide** or **Customers in Germany**, and I can also delete a customer. I _cannot_, however, edit or create a customer. Changes are not saved when editing, and I get client-side errors when trying to create a customer.
5. In your step 7, you tell us to click on Customers Worldwide and view the Network tab to view the JSON response from the GET request, but the response shows **_"Failed to load response data: No data found for resource with given identifier"_** When I click on the Headers tab under Network, the Response Headers shows a content-type of **text/html; charset=utf-8** instead of JSON.

**Comment by pmarangoni** [comment]
Just to add, this is a screenshot when trying to create a new customer. I enter values for each field, and after submitting the form (by clicking Create Customer), the page reloads with errors displayed under Customer Id and Company Name.
![Screenshot 2024-07-23 135654](https://github.com/user-attachments/assets/9e4c0b30-d4de-40a8-9599-b74c20a75a4d)

**Comment by pmarangoni** [comment]
A little help please.

**Comment by markjprice** [comment]
I have updated the online section with new instructions and made the same changes to the `PracticalApps` solution code in the `Northwind.WebApi`, `Northwind.Blazor`, and `Northwind.Blazor.Wasm` projects. This should fix the problems.
https://github.com/markjprice/cs12dotnet8/blob/main/docs/ch15-blazor-wasm.md

**Comment by pmarangoni** [comment]
Without knowing specifically what was changed I will need to start over.

**Comment by pmarangoni** [comment]
> 4. Navigate to https://localhost:5161/.

I get this:
![image](https://github.com/user-attachments/assets/e77a8d71-662e-49d9-8c86-10a93918dde8)

**Comment by pmarangoni** [comment]
**Switching to the WebAssembly component**

I was getting weird errors trying to get this to work.

> Failed to start platform. Reason: TypeError: Cannot read properties of undefined (reading 'AspNetCore')

I found that others had similar issues and followed their advice and got it to work. Basically, I closed Visual Studio, then deleted the folders underneath the Northwind.Blazor.Wasm project's **obj** and **bin** folders. Restarted Visual Studio and it worked fine.

This is where I found the advice:
https://github.com/dotnet/aspnetcore/issues/50755

**Comment by markjprice** [comment]
I agree with you. To me, Blazor is still an "unstable" platform both in that it has more bugs than alternatives like MVC and that it is still evolving. 

.NET 8 introduced huge changes in Blazor that meant the way I had designed my step-by-step instructions for the .NET 7 edition then broke with .NET 8 in ways that I discovered couldn't be easily fixed. When I first looked at the issues readers like you were having I tried to find some simpler fixes but ended up having to almost rewrite that whole section. 

Blazor feels too much like the old Web Forms (`.aspx`) which tried to hide too much and ended up building a massive mess on top of web development. 

I'm sorry I wasn't able to provide a solution earlier. I appreciate how frustrating that must have been for you and others. Thank you for your patience.

p.s. I will add the information you found about the `Failed to start platform.` issue to the page too.
